### PR TITLE
[FLINK-13828][configuration] Deprecate ConfigConstants.LOCAL_START_WEBSERVER

### PR DESCRIPTION
**Pull request by TisonKun**
## What is the purpose of the change

Maybe backport to 1.9.1 because I heard this issue from a user of 1.9.0

Exactly ConfigConstants.LOCAL_START_WEBSERVER has no power any more but we don't mark it as deprecated or give an investigate to revive it. Since it is part of public interface we cannot remove it in minor version but deprecate it is necessary.

## Brief change log

- Remove use points of `ConfigConstants.LOCAL_START_WEBSERVER`
- Deprecate `ConfigConstants.LOCAL_START_WEBSERVER`
- hotfix for test style in `WebFrontendITCase`

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes, mark a stale field as deprecated.)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

cc @twalthr

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java [hunk 1]
```diff
@@ -1866,6 +1866,10 @@
 	@Deprecated
 	public static final int DEFAULT_LOCAL_NUMBER_RESOURCE_MANAGER = 1;
 
+	/**
+	 * Ineffective any more. Will be removed in 2.0.
```
Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java]:
```suggestion
	 * @deprecated Has no effect; the web-server is always started. Will be removed in 2.0.
```

File: flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/WebFrontendITCase.java [hunk 2]
```diff
@@ -209,45 +198,37 @@ public void getLogAndStdoutFiles() throws Exception {
...
 
-			Map<String, String> conf = WebMonitorUtils.fromKeyValueJsonArray(config);
-			assertEquals(
-				CLUSTER_CONFIGURATION.getString(ConfigConstants.LOCAL_START_WEBSERVER, null),
-				conf.get(ConfigConstants.LOCAL_START_WEBSERVER));
-		} catch (Exception e) {
-			e.printStackTrace();
-			fail(e.getMessage());
-		}
+		String expected = CLUSTER_CONFIGURATION.getString(TaskManagerOptions.MANAGED_MEMORY_SIZE);
```
Comment (zentol) [inline flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/WebFrontendITCase.java]: Please add a comment to the setup section pointing out that setting this value is required for the validity of some tests.

File: flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/WebFrontendITCase.java [hunk 3]
```diff
@@ -209,45 +198,37 @@ public void getLogAndStdoutFiles() throws Exception {
...
+		String actual = conf.get(TaskManagerOptions.MANAGED_MEMORY_SIZE.key());
+
+		assertNotNull(actual);
```
Comment (zentol) [inline flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/WebFrontendITCase.java]: seems unnecessary, assertEquals also handles this case.

Comment (TisonKun) [reply flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/WebFrontendITCase.java]: I think this prevent the case `null == null` but if we add comment above and assume we always set `TaskManagerOptions.MANAGED_MEMORY_SIZE` I agree with can remove this one.

**Comment by zentol** [comment]
@TisonKun Could you squash your last commit (61306fa) into the first one (933c1ed)? I would then merge it afterwards.

**Comment by TisonKun** [comment]
@zentol Thanks for your review. Updated.

**Pull request merged**